thorcast/geocodex.py

```diff
@@ -0,0 +1,112 @@
+"""
+Database interface utility
+
+The database is used to store coordinates of previously requested
+locations. This is implemented to prevent going over GCP's free
+allotment of 40k free geocode requests/month.
+"""
+
+import datetime
+import os
+
+import sqlalchemy
+
+def gen_conn_str(username, password, host, port, db):
+
+    return f'postgresql://{username}:{password}@{host}:{port}/{db}'
+
+
+class Geocodex(object):
```

Comment:
Based on the comments of what this class is doing, you're probably over doing it a bit here.
There's probable a disk cache decorator out there somewhere or you could make a simple one yourself.

The use of such a decorator would be:
```python
@disk_cache('/path/to/writable/cache/dir', ttl=86400)
def fetch_me_long_time(x, y):
   internal.request(x,y)
```

There's a good example of making your own decorator here: https://realpython.com/primer-on-python-decorators/

Reply:
Are you saying I don't need to use Redis?

Reply:
er, this is the Postgres wrapper class.